components/brave_wallet_ui/common/hardware/ledgerjs/ledger_bridge_keyring.ts

```diff
@@ -0,0 +1,109 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { LEDGER_HARDWARE_VENDOR } from 'gen/brave/components/brave_wallet/common/brave_wallet.mojom.m.js'
+import { BraveWallet } from '../../../constants/types'
+// import { HardwareKeyring } from '../interfaces'
+import { getLocale } from '../../../../common/locale'
+import { HardwareVendor } from '../../api/hardware_keyrings'
+import {
+  HardwareOperationResult
+} from '../types'
+import {
+  LEDGER_BRIDGE_URL,
+  LedgerCommand,
+  UnlockResponse,
+  LedgerFrameCommand,
+  LedgerBridgeErrorCodes
+} from './ledger-messages'
+import { LedgerTrustedMessagingTransport } from './ledger-trusted-transport'
+
+// storybook compiler thinks `randomUUID` doesn't exist
+const randomUUID = () => (window.crypto as Crypto & { randomUUID: () => string }).randomUUID()
```

Comment:
nit can we refactor this logic into a util function somewhere? I see it very commonly used and would be good to not need to rewrite it everywhere

Reply:
Really all we're trying to do is use the built in [window.crypto.randomUUID](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID) function, but for whatever reason the compiler thinks that doesn't exist, so we have do go through hoops to escape the error it raises.

In this case the `randomUUID` function is used by this new LedgerBridgeKerying class, which is now the parent class of both the EthereumLedgerBridgeKeyring and SolanaLedgerBridgeKeyring class.  Because of that refactor, we aren't defining this function of each of the child classes and I believe we have removed reuse in that way.

Reply:
> Really all we're trying to do is use the built in [window.crypto.randomUUID](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID) function, but for whatever reason the compiler thinks that doesn't exist, so we have do go through hoops to escape the error it raises.

Ahh I bet I know why this is. Are you using VS Code? If so I think what's happening is the typesystem built into VS Code is automatically imported into every TS file for some reason and so what happens is within VSCode the types from lib.dom.d.ts are what are being used, but when the storybook compiler is ran it's not able to detect these types. I believe if you add these types in explicitly to the files where this logic is used you shouldn't encounter the issue anymore.

I remember I ran into this issue in the past when I was working with [JsonWebKey](https://github.com/microsoft/TypeScript/blob/main/lib/lib.dom.d.ts#L633) and I kept getting compiler errors because of it.

Reply:
Opened the issue here: https://github.com/brave/brave-browser/issues/24764